### PR TITLE
Upgrade to 2.3.1 sdk

### PR DESCRIPTION
**Pull request by thinkocapo**
2.1.4 > 2.3.1

## SDK Roadmap
https://github.com/getsentry/sentry-android might get archived in Sept 2020

Android's sdk will be under sentry-java. **Unified API** alpha version today Sept,2020

so can make better use of gradle and modules. sentry-android is an integration. shared code base. they share `sentry/core`

be on the lookout for any migration document that comes up, for newer sdk.

ReleaseHealth will be enabled by default.

## Testing

[unhandled - arithmetic exception](https://sentry.io/organizations/testorg-az/discover/paranoid-android:de916518195a48ed98658202a54b99ea/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&query=&sort=-timestamp&statsPeriod=24h&widths=-1&widths=-1&widths=-1&widths=-1&widths=-1)

[Handled - ArrayIndexOutOfBounds](https://sentry.io/organizations/testorg-az/discover/paranoid-android:229d25dd8494436484858e20fdf209c6/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&query=&sort=-timestamp&statsPeriod=24h&widths=-1&widths=-1&widths=-1&widths=-1&widths=-1)

[Message Capture](https://sentry.io/organizations/testorg-az/discover/paranoid-android:0c68781149ac40e875e917f4dde8ef00/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&query=&sort=-timestamp&statsPeriod=24h&widths=-1&widths=-1&widths=-1&widths=-1&widths=-1)

[native crash c++](https://sentry.io/organizations/testorg-az/discover/paranoid-android:ae713e3dc6f64dc1af145a371112711b/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&query=&sort=-timestamp&statsPeriod=24h&widths=-1&widths=-1&widths=-1&widths=-1&widths=-1)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,7 +60,7 @@ Additional documentation:
 
 6. `make all`
 
-7. Android Studio install Android NDK in Preferences > System & Behavior > System Settings > Android SDK > SDK Tools > select NDK for download
+7. Android Studio install Android NDK in Preferences > System & Behavior > System Settings > Android SDK > SDK Tools > select NDK for download, and CMake
```
Comment (thinkocapo) [inline README.md:63]: @marandaneto do you know if we need CMake? I saw I had it downloaded on my other computer's AndroidStudio so I also checked it off when setting up new computer with Android Studio

File: app/build.gradle [hunk 2]
```diff
@@ -44,14 +44,15 @@ android {
     lintOptions {
         abortOnError false
     }
+    ndkVersion '21.3.6528147'
```
Comment (thinkocapo) [inline app/build.gradle:47]: this appeared on me. Confirmed with Manoel it's not required.

File: README.md (hunk 1 above)
Comment (marandaneto) [reply README.md:63]: yes, sdk build-tools, ndk, cmake, sdk platform-tools, sdk tools, haxm recommended, lldb recommended to debug ndk

File: app/build.gradle (hunk 2 above)
Comment (marandaneto) [reply app/build.gradle:47]: not really required, the android gradle plugin already gets the latest version downloaded/installed in your machine that is compatible with your project, although pinning a version is also good for people that likes to play safe

**Review by marandaneto — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (thinkocapo) [reply README.md:63]: thanks, i'm updating this.

**Pull request merged**